pkgs/servers/onlyoffice-documentserver/default.nix

```
@@ -17,11 +17,22 @@ let
     pname = "onlyoffice-documentserver";
     version = "7.5.1";
 
-    src = fetchurl {
+    src =  let
+       onlyoffice-documentserver-amd64 = {
       url = "https://github.com/ONLYOFFICE/DocumentServer/releases/download/v${lib.concatStringsSep "." (lib.take 3 (lib.splitVersion version))}/onlyoffice-documentserver_amd64.deb";
       sha256 = "sha256-191PYpxs/TbVXoBPHvuyTp81ZMtw1YaFznY1hUSbh+0=";
     };
 
+    onlyoffice-documentserver-arm64 = {
+      url = "https://github.com/ONLYOFFICE/DocumentServer/releases/download/v${lib.concatStringsSep "." (lib.take 3 (lib.splitVersion version))}/onlyoffice-documentserver_arm64.deb";
+      sha256 = "sha256-j16LrYDq/7kuJWAmCb8BTJ6S9LyJ092cwRemDic7EOs=";
+    };
+    in fetchurl (if stdenv.hostPlatform.isAarch64
+      then onlyoffice-documentserver-arm64
+      else if stdenv.hostPlatform.isx86_64
+      then onlyoffice-documentserver-amd64
+      else throw "unsupported platform");
```

```suggestion
    src = fetchurl ({
      "aarch64-linux" = {
        url = "https://github.com/ONLYOFFICE/DocumentServer/releases/download/v${version}/onlyoffice-documentserver_arm64.deb";
        sha256 = "sha256-j16LrYDq/7kuJWAmCb8BTJ6S9LyJ092cwRemDic7EOs=";
      };
      "x86_64-linux" = {
        url = "https://github.com/ONLYOFFICE/DocumentServer/releases/download/v${version}/onlyoffice-documentserver_amd64.deb";
        sha256 = "sha256-191PYpxs/TbVXoBPHvuyTp81ZMtw1YaFznY1hUSbh+0=";
      }
    }.${stdenv.hostPlatform.system} or throw "unsupported platform");
```

I don't think we need the extra concatStringsSep anymore. All versions are now 3 digits.

---

the throw was returning this issue:
```console
 error: attempt to call something which is not a function but a set

       at /home/eymeric/tmp/nixpkgs2/pkgs/servers/onlyoffice-documentserver/default.nix:20:21:

           19|
           20|     src = fetchurl ({
             |                     ^
           21|       "aarch64-linux" = {
```
so I just removed it because I think not throwing would do the same thing because fetchurl will throw because we give him a string and not a set

---

> the throw was returning this issue:

It's because of missing the parentheses. It should be `or (throw "whatever")` not `or throw "whatever"`